### PR TITLE
feat: [PL-42000]: add index to support deletion for a notification for an accountId

### PR DESCRIPTION
**Pull request by sahibaHarness**
## Describe your changes
As part of data cleanup activity Notification related data gets clean up when an account is deleted. In this we remove all the data in the collection for given accountId.
As there is no index available on accountId it does a COLLSCAN.

Creating an index on accountId and createdAt so that accounts can be filtered on accountID fro deletion and further more can be sorted on timestamp in future.

https://harness.slack.com/archives/C047UK9H3NE/p1697448247729709

MATCHING VALUES: 17493
TOTAL NUMBER OF RECORDS: 19889
Before adding this index:
Pasted Graphic
post adding index:
Pasted Graphic 1

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54184)

### CODE REVIEW

**Comment by sahibaHarness** [comment]
trigger uts

File: platform-service/modules/notification-service/src/main/java/io/harness/notification/entities/Notification.java [hunk 1]
```diff
@@ -54,6 +54,11 @@ public static List<MongoIndex> mongoIndexes() {
                  .field(NotificationKeys.shouldRetry)
                  .field(NotificationKeys.retries)
                  .build())
+        .add(CompoundMongoIndex.builder()
+                 .name("notification_account_idx")
+                 .field(NotificationKeys.accountIdentifier)
+                 .field(NotificationKeys.createdAt)
+                 .build())
```
Comment (bhavya181) [inline platform-service/modules/notification-service/src/main/java/io/harness/notification/entities/Notification.java:61]: there is no need to create index on accountId and createdAt, there is no query to list notifications right now, so just put an FDIndex on accountId for now, if we add some query on timestamp in future then that time we will create index on timestamp.

**Review by bhavya181 — CHANGES_REQUESTED** [review]

File: platform-service/modules/notification-service/src/main/java/io/harness/notification/entities/Notification.java (hunk 1 above)
Comment (sahibaHarness) [reply platform-service/modules/notification-service/src/main/java/io/harness/notification/entities/Notification.java:60]: ack

**Comment by sahibaHarness** [comment]
trigger uts

**Review by ankit-harness — APPROVED** [review]

**Pull request merged**